src/Trigger/RabbitMQListener.cs

```diff
@@ -139,12 +144,77 @@ internal void RepublishMessages(BasicDeliverEventArgs ea)
             }
         }
 
+        public string Id
+        {
+            get
+            {
+                return $"RabbitMQTrigger-{_queueName}".ToLower();
+            }
+        }
+
         private void ThrowIfDisposed()
         {
             if (_disposed)
             {
                 throw new ObjectDisposedException(null);
             }
         }
+
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            return await GetMetricsAsync();
+        }
+
+        public async Task<RabbitMQTriggerMetrics> GetMetricsAsync()
+        {
+            return new RabbitMQTriggerMetrics
+            {
+                QueueLength = _queueInfo.MessageCount,
+            };
+        }
+
+        ScaleStatus IScaleMonitor.GetScaleStatus(ScaleStatusContext context)
+        {
+            ScaleStatus status = new ScaleStatus
+            {
+                Vote = ScaleVote.None,
+            };
+
+            const int NumberOfSamplesToConsider = 5;
+
+            List<RabbitMQTriggerMetrics> metrics = (context as ScaleStatusContext<RabbitMQTriggerMetrics>)?.Metrics?.ToList();

```

Comment:
I realized I had a bug in the QueueListener code where you copied this from. See the latest implementation. Issue was these apis handle both generic/non-generic calls, and there was a casting bug making my impl. not work in some cases. Recommend adding tests for both generic and non-generic for these interfaces as I've done (https://github.com/Azure/azure-webjobs-sdk/pull/2315/files#diff-18dc160d04acf1baa68eee0610eb3d74R101)